diff --git a/packages/flutter_app_packager/lib/src/makers/appimage/make_appimage_config.dart b/packages/flutter_app_packager/lib/src/makers/appimage/make_appimage_config.dart
--- a/packages/flutter_app_packager/lib/src/makers/appimage/make_appimage_config.dart
+++ b/packages/flutter_app_packager/lib/src/makers/appimage/make_appimage_config.dart
@@ -38,6 +38,7 @@ class MakeAppImageConfig extends MakeConfig {
     this.actions = const [],
     this.include = const [],
     this.startupNotify = true,
+    this.startupWmClass,
     this.genericName = 'A Flutter Application',
   });
@@ -49,6 +50,7 @@ class MakeAppImageConfig extends MakeConfig {
       categories: (map['categories'] as List<dynamic>? ?? []).cast<String>(),
       startupNotify: map['startup_notify'] as bool? ?? false,
+      startupWmClass: map['startup_wm_class'] as String?,
       genericName: map['generic_name'] as String? ?? 'A Flutter Application',
       actions: (map['actions'] as List? ?? [])
@@ -65,6 +67,7 @@ class MakeAppImageConfig extends MakeConfig {
   final List<String> categories;
   final List<AppImageAction> actions;
   final bool startupNotify;
+  final String? startupWmClass;
   final String genericName;
   final String displayName;
   final List<String> include;
@@ -77,6 +80,7 @@ class MakeAppImageConfig extends MakeConfig {
       'Icon': appName,
       'Type': 'Application',
       'StartupNotify': startupNotify ? 'true' : 'false',
+      if (startupWmClass != null) 'StartupWMClass': startupWmClass,
       if (categories.isNotEmpty) 'Categories': categories.join(';'),
       if (keywords.isNotEmpty) 'Keywords': keywords.join(';'),
       if (this.actions.isNotEmpty)
diff --git a/packages/flutter_app_packager/lib/src/makers/deb/make_deb_config.dart b/packages/flutter_app_packager/lib/src/makers/deb/make_deb_config.dart
--- a/packages/flutter_app_packager/lib/src/makers/deb/make_deb_config.dart
+++ b/packages/flutter_app_packager/lib/src/makers/deb/make_deb_config.dart
@@ -130,6 +130,7 @@ class MakeDebConfig extends MakeLinuxPackageConfig {
     required this.installedSize,
     required this.maintainer,
     this.startupNotify = true,
+    this.startupWmClass,
     this.essential = false,
     List<String>? postinstallScripts,
     List<String>? postuninstallScripts,
@@ -217,6 +218,7 @@ class MakeDebConfig extends MakeLinuxPackageConfig {
       genericName: map['generic_name'],
       startupNotify: map['startup_notify'],
+      startupWmClass: map['startup_wm_class'],
       installedSize: map['installed_size'],
       icon: map['icon'],
     );
@@ -232,6 +234,7 @@ class MakeDebConfig extends MakeLinuxPackageConfig {
   String? genericName;
   bool? startupNotify;
+  String? startupWmClass;
   List<String>? coAuthors;
   List<String>? dependencies;
   List<String>? buildDependenciesIndep;
@@ -309,6 +312,7 @@ class MakeDebConfig extends MakeLinuxPackageConfig {
         'Keywords': keywords != null && keywords!.isNotEmpty
             ? '${keywords!.join(';')};'
             : null,
         'StartupNotify': startupNotify,
+        'StartupWMClass': startupWmClass,
       }..removeWhere((key, value) => value == null),
     };
   }
diff --git a/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart b/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart
--- a/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart
+++ b/packages/flutter_app_packager/lib/src/makers/rpm/make_rpm_config.dart
@@ -7,6 +7,7 @@ class MakeRPMConfig extends MakeConfig {
     // Desktop file
     required this.displayName,
     this.startupNotify = true,
+    this.startupWmClass,
     this.actions,
     this.categories,
     this.genericName,
@@ -42,6 +43,7 @@ class MakeRPMConfig extends MakeConfig {
       icon: json['icon'] as String?,
       genericName: json['generic_name'] as String?,
       startupNotify: json['startup_notify'] as bool?,
+      startupWmClass: json['startup_wm_class'] as String?,
       keywords: (json['keywords'] as List<dynamic>?)?.cast<String>(),
       supportedMimeType:
@@ -73,6 +75,7 @@ class MakeRPMConfig extends MakeConfig {
   String? icon;
   String? genericName;
   bool? startupNotify;
+  String? startupWmClass;
   List<String>? keywords;
   List<String>? supportedMimeType;
@@ -166,6 +169,7 @@ class MakeRPMConfig extends MakeConfig {
         'Keywords': keywords != null && keywords!.isNotEmpty
             ? '${keywords!.join(';')};'
             : null,
         'StartupNotify': startupNotify,
+        'StartupWMClass': startupWmClass,
       }..removeWhere((key, value) => value == null),
     };
   }
